### PR TITLE
feat(proxy): implement custom proxy agent support in BitGoAPI

### PR DESCRIPTION
**Pull request by ayush-9**
TICKET: CE-4801

BREAKING CHANGE
- Instead of getting the proxy url, now he caller should just pass in an instance of the proxy-agent they wish to use in the sdk-api, using a new param named `customProxyAgent`
- Making bitgo/express pass in the proxy agent by default

### CODE REVIEW

**Review by OttoAllmendinger — DISMISSED** [review]

File: modules/sdk-api/src/types.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { EnvironmentName, IRequestTracer, V1Network } from '@bitgo/sdk-core';
 import { ECPairInterface } from '@bitgo/utxo-lib';
+import { Agent } from 'http';
```
Comment (bitgoAaron) [inline modules/sdk-api/src/types.ts:3]: import as `type`

**Review by bitgoAaron — APPROVED** [review]

**Review by bitgoAaron — CHANGES_REQUESTED** [review]
please import as type where appropriate to avoid clients having to parse the type

File: modules/express/src/clientRoutes.ts [hunk 2]
```diff
@@ -1213,12 +1214,19 @@ function prepareBitGo(config: Config) {
...
       env,
       customRootURI: customRootUri,
       customBitcoinNetwork,
       accessToken,
       userAgent,
+      ...(useProxyUrl && {
+        customProxyAgent: new ProxyAgent({
+          getProxyForUrl: () => useProxyUrl,
+        }),
+      }),
```
Comment (bitgoAaron) [inline modules/express/src/clientRoutes.ts:1229]:
```suggestion
      ...(useProxyUrl ? {
        customProxyAgent: new ProxyAgent({
          getProxyForUrl: () => useProxyUrl,
        } : {} ),
      }),
```

**Review by bitgoAaron — APPROVED** [review]

**Comment by ayush-9** [comment]
closing this pr as the github is not able to load changes

**Pull request closed without merging**